runtime/parachains/src/inclusion.rs

```diff
@@ -565,17 +588,28 @@ impl<T: Config> Pallet<T> {
 								},
 							}
 
-							for (bit_idx, _) in candidate
+							let mut backer_idx_and_signature =
+								Vec::<(ValidatorIndex, ValidatorSignature)>::with_capacity(
+									backed_candidate.validator_indices.count_ones(),
+								);
+							for ((bit_idx, _), attestation) in backed_candidate
 								.validator_indices
 								.iter()
 								.enumerate()
 								.filter(|(_, signed)| **signed)
+								.zip(backed_candidate.validity_votes.iter())
 							{
-								let val_idx =
-									group_vals.get(bit_idx).expect("this query done above; qed");
+								let val_idx = group_vals
+									.get(bit_idx)
+									.expect("this query succeeded above; qed");
+
+								backer_idx_and_signature
+									.push((*val_idx, attestation.signature().clone()));
```

Comment:
it should probably store the attestation type as well, re: https://github.com/paritytech/polkadot/pull/3924/files#r720683514